src/main/java/com/aws/greengrass/componentmanager/builtins/GreengrassRepositoryDownloader.java

```diff
@@ -112,6 +116,8 @@ private Long getDownloadSizeWithoutRetry() throws InterruptedException, PackageD
                             DeploymentErrorCode.GREENGRASS_ARTIFACT_SIZE_NOT_FOUND);
                 }
                 return length;
+            } else if (RetryUtils.retryErrorCodes(responseCode)) {
+                throw new RetryableCloudDeploymentException("Throwing on 500 error: " + responseCode);
```

Comment:
nit: the message here is not accurate since it's not just 500 you are retrying on.

Reply:
nit: i think you can also improve other exception messages as well

Reply:
Re-opening this comment since I don't see it addressed, let's not say 500 since we know it could be other error codes, and I agree that the error message could be better